naga/src/front/spv/mod.rs

```diff
@@ -3943,7 +3960,84 @@ impl<I: Iterator<Item = u32>> Frontend<I> {
                     );
                     emitter.start(ctx.expressions);
                 }
-                _ => return Err(Error::UnsupportedInstruction(self.state, inst.op)),
+                Op::AtomicIIncrement => {
+                    inst.expect(6)?;
+                    let start = self.data_offset;
+                    let span = self.span_from_with_op(start);
+                    let result_type_id = self.next()?;
+                    let result_id = self.next()?;
+                    let pointer_id = self.next()?;
+                    let scope_id = self.next()?;
+                    let memory_semantics_id = self.next()?;
+                    // Store the op for a later pass where we "upgrade" the pointer type
+                    let atomic = AtomicOp {
+                        instruction: AtomicOpInst::AtomicIIncrement,
+                        result_type_id,
+                        result_id,
+                        pointer_id,
+                        scope_id,
+                        memory_semantics_id,
+                    };
+                    self.lookup_atomic.insert(pointer_id, atomic);
+
+                    log::trace!("\t\t\tlooking up expr {:?}", pointer_id);
+
+                    let (p_lexp_handle, p_lexp_ty_id) = {
+                        let lexp = self.lookup_expression.lookup(pointer_id)?;
+                        let handle = get_expr_handle!(pointer_id, &lexp);
+                        (handle, lexp.type_id)
+                    };
+                    log::trace!("\t\t\tlooking up type {pointer_id:?}");
+                    let p_ty = self.lookup_type.lookup(p_lexp_ty_id)?;
+                    let p_base_ty = if let Some(id) = p_ty.base_id {
+                        log::trace!("\t\t\tlooking up base type {id:?} of {p_ty:?}");
+                        self.lookup_type.lookup(id)?
+                    } else {
+                        log::warn!("\t\t\ttype {p_ty:?} has no base id");
+                        p_ty
+                    };
```

Comment:
I think it should be fine to just return an error if `p_ty.base_id` is not `Some`, because the only way that `Frontend::parse_type_pointer` is ever going to put a `naga::TypeInner::Pointer` type in `self.lookup_type` in the first place is if it's got a base type.

(I assume the overall effort here is not aiming to add support for atomic operations on images - is that correct?)

Reply:
> I assume the overall effort here is not aiming to add support for atomic operations on images - is that correct?
Your assumption is correct in that I hadn't even considered it. I think that makes it out of scope for my project, but if I need it down the line you bet I'll try to cut a PR for it later :)